spring-kafka/src/main/java/org/springframework/kafka/retrytopic/ExceptionBasedDltDestination.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.retrytopic;
+
+/**
+ * Annotation allowing to specify additional DLT which will be chosen when message
+ * processing caused the configured exception to be thrown.
+ *
+ * @author Adrian Chlebosz
+ * @see org.springframework.kafka.retrytopic.ExceptionBasedDltRouting
+ * @since 3.2.0
+ */
+public @interface ExceptionBasedDltDestination {
+
+	/**
+	 * Suffix extension used when constructing the name for the new DLT. It is placed
+	 * before the main suffix configured through the
+	 * ${@link org.springframework.kafka.annotation.RetryableTopic#dltTopicSuffix()}, so the
+	 * final name is the product of these two.
+	 *
+	 * @return the configured suffix extension
+	 */
+	String suffix();
```

Comment:
I wonder if this is really required and can be just empty by default...

Also: no blank lines in the method Javadocs: https://github.com/spring-projects/spring-framework/wiki/Code-Style#javadoc-formatting

Reply:
From my point of view having the `suffix` required makes this API harder to misuse by design. `exceptions` is required, but when someone will be allowed to specify exceptions without the `suffix`, then it means that the general purpose DLT will be used even for this exc. I'm not sure when it could be an intended usage